toxic_collection.go

```diff
@@ -96,20 +96,31 @@ func (c *ToxicCollection) AddToxicJson(data io.Reader) (*toxics.ToxicWrapper, er
 		return nil, joinError(err, ErrBadRequestBody)
 	}
 
-	switch strings.ToLower(wrapper.Stream) {
-	case "downstream":
-		wrapper.Direction = stream.Downstream
-	case "upstream":
-		wrapper.Direction = stream.Upstream
-	default:
-		return nil, ErrInvalidStream
+	if toxics.New(wrapper) == nil {
+		return nil, ErrInvalidToxicType
 	}
-	if wrapper.Name == "" {
-		wrapper.Name = fmt.Sprintf("%s_%s", wrapper.Type, wrapper.Stream)
+
+	if wrapper.PairedToxic == nil {
+		switch strings.ToLower(wrapper.Stream) {
+		case "downstream":
+			wrapper.Direction = stream.Downstream
+		case "upstream":
+			wrapper.Direction = stream.Upstream
+		default:
+			return nil, ErrInvalidStream
+		}
+	} else {
+		wrapper.Stream = "both"
```

Comment:
I'm not sure what the behaviour should be here since there's only one option for bidirectional toxics.
Right now `stream` can be set to anything and it will just be changed to `both` by the server.